### DISCUSSION TITLE
What info does GitHub actually provide?

### DISCUSSION THREAD

**Discussion by Andre601** [Readme-Workflows/recent-activity#16] [Q&A] [answered]
This is something I currently wonder. What info does GitHub provide for the Action to use?
Perhaps there are some other options/features that people may be interested in using?

**Comment by abhijoshi2k** [comment, marked as answer]
https://api.github.com/users/{USERNAME}/events/public
Send a GET request or simply put this in browser (make sure you have JSON formatter extension in browser).
That is the format we get info in.
Docs : https://docs.github.com/en/developers/webhooks-and-events/github-event-types

**Reply by abhijoshi2k** [reply]
I hope this was what you asked.

**Reply by Andre601** [reply]
Btw. It looks like Org accounts also have recent activity info. It's basically a mirror of the single user one, but with mixed users/actors.
So if we want could we have an example-repository with a working GitHub Action on this Organisation.

**Reply by abhijoshi2k** [reply]
> Btw. It looks like Org accounts also have recent activity info. It's basically a mirror of the single user one, but with mixed users/actors.
> So if we want could we have an example-repository with a working GitHub Action on this Organisation.

But for that authors will be different right?
So we'll have to make changes in workflow to include authors too.

**Reply by Andre601** [reply]
Does it really matter to show who made this action?

All I want to say is, that we could have an example repository to showcase the action without the need of suffering a repo on a personal account for this.

**Reply by abhijoshi2k** [reply]
Yes we can try it I guess. @PuneetGopinath can you do this part?

**Reply by PuneetGopinath** [reply]
> Yes we can try it I guess. @PuneetGopinath can you do this part?

Yes

**Reply by PuneetGopinath** [reply]
I simply created a repo [recent-activity-template](//github.com/PuneetGopinath/recent-activity-template) for that

**Reply by abhijoshi2k** [reply]
> I simply created a repo [recent-activity-template](//github.com/PuneetGopinath/recent-activity-template) for that
> Try to fix #23 for that repo to work

But why do we need v1 to work?
As of now, the workflow is working properly on main branch.

**Reply by PuneetGopinath** [reply]
I have invited you both as collaborator there, go ahead and submit PRS to that repo.

**Reply by abhijoshi2k** [reply]
> I have invited you both as collaborator there, go ahead and submit PRS to that repo.

I'd say let's just keep that readme simple.
That'll use only our workflow.

**Comment by PuneetGopinath** [comment]
> Perhaps there are some other options/features that people may be interested in using?

Yep, some other events I know are:
1. ForkEvent
2. CreateEvent
3. ReleaseEvent
4. PushEvent

**Reply by abhijoshi2k** [reply]
Yeah. Let's make a list of events that are worth adding and add them to the workflow.

**Reply by abhijoshi2k** [reply]
> > Perhaps there are some other options/features that people may be interested in using?
> 
> Yep, some other events I know are:
> 
> 1. ForkEvent
> 2. CreateEvent
> 3. ReleaseEvent
> 4. PushEvent

I think we should add these 4 events.
Also, maybe we should disable the push events by default as we push changes much more often than other events.

**Reply by Andre601** [reply]
Yeah. IMO should pushes be left out completely since from what I know does f.e. a merge also count as a push, no?
Or a release also triggers a push, as you may have noticed once yourself with workflows triggering on push.

For me PushEvent seems pointless to have as it often is just a byproduct of another event.

**Reply by abhijoshi2k** [reply]
> Yeah. IMO should pushes be left out completely since from what I know does f.e. a merge also count as a push, no?
> Or a release also triggers a push, as you may have noticed once yourself with workflows triggering on push.
> 
> For me PushEvent seems pointless to have as it often is just a byproduct of another event.

Merging creates a pullRequestEvent and release too has its own event.
I will still go through docs and API to confirm this.

**Reply by PuneetGopinath** [reply]
No, a merge doesn't count as push, a push means pushing a commit

**Reply by Andre601** [reply]
I just learned from personal experience that when I made a new release, that my workflow, which was set to trigger on push, activated from this because the release triggered a new push since a Tag was generated.

**Reply by PuneetGopinath** [reply]
> I just learned from personal experience that when I made a new release, that my workflow, which was set to trigger on push, activated from this because the release triggered a new push since a Tag was generated.

I think because you tagged, so the latest commit in that tag just ran the workflow

**Reply by abhijoshi2k** [reply]
I think @Andre601 is right. Push event does fire when we merge/release. I just checked my events by sending a GET request at:
https://api.github.com/users/{USERNAME}/events/public

**Reply by PuneetGopinath** [reply]
Ok, maybe right

**Reply by abhijoshi2k** [reply]
Also, I just now released v1.2.2
But I don't see a release event. Any idea why that would happen? @PuneetGopinath @Andre601

**Reply by abhijoshi2k** [reply]
Even CreateEvent fires when you create a branch.
Should we restrict it to creating repo? Or should we add creating branch as well?

**Reply by Andre601** [reply]
Are there any more infos regarding this? Like is there specific info to distinguish both from each other? In such a case could we offer them both as separate event.

**Reply by PuneetGopinath** [reply]
> Also, I just now released v1.2.2
> But I don't see a release event. Any idea why that would happen? @PuneetGopinath @Andre601

Haha, GitHub actions released that not you.
That's because GitHub actions drafted it.

**Reply by Andre601** [reply]
> Also, I just now released v1.2.2
> But I don't see a release event. Any idea why that would happen? @PuneetGopinath @Andre601

My guess is, because it's made by github-actions and not you.

**Reply by abhijoshi2k** [reply]
> > Also, I just now released v1.2.2
> > But I don't see a release event. Any idea why that would happen? @PuneetGopinath @Andre601
> 
> My guess is, because it's made by github-actions and not you.

Okay. I get it now.

**Reply by abhijoshi2k** [reply]
> Are there any more infos regarding this? Like is there specific info to distinguish both from each other? In such a case could we offer them both as separate event.

Yes we can distinguish. There are some properties that change.

**Reply by PuneetGopinath** [reply]
Do chat in discord
